test/analysis_tests.py

```diff
@@ -1009,6 +1035,414 @@ def test_loss(self):
+
+		yptf = tf.constant(np.concatenate([y_pred,L_elements],axis=-1),
+			dtype=tf.float32)
+		yttf = tf.constant(y_true,dtype=tf.float32)
+		loss = loss_class.loss(yttf,yptf)
+
+		self.assertAlmostEqual(np.sum(loss.numpy()),scipy_nlp,places=4)
+
+class FullCovarianceAPTLossTests(unittest.TestCase):
+
+	def setUp(self):
+		# Set up a random seed for consistency
+		np.random.seed(2)
+
+	def test_sanity_loss(self):
+
+        # if prior = proposal, check that produces the same as without APT
+        # modification
+		dim = 2
+		mu_prior = np.zeros(dim)
+		prec_prior = np.diag(np.ones(dim) * 4)
+		mu_prop = mu_prior
+		prec_prop = prec_prior
+
+		gaussian_loss = Analysis.loss_functions.FullCovarianceLoss(dim)
+		snpe_c_loss = Analysis.loss_functions.FullCovarianceAPTLoss(dim,
+			mu_prior, prec_prior, mu_prop,prec_prop)
+
+		batch_size = int(1e4)
+
+		# produce output = y_pred, L_mat_elements in flattened sequence
+		len_L_mat_elements = int(dim*(dim+1)/2)
+		outputs = np.concatenate(
+			[np.random.normal(size=(batch_size, dim)),
+				np.zeros((batch_size, len_L_mat_elements))], axis=-1
+		)
+
+		truth = np.random.normal(size=(batch_size, dim))
+
+		truth = tf.constant(truth,dtype=tf.float32)
+		outputs = tf.constant(outputs,dtype=tf.float32)
+
+		print(truth.shape)
+
+		np.testing.assert_almost_equal(gaussian_loss.loss(truth,outputs).numpy(),
+			snpe_c_loss.loss(truth,outputs).numpy())
+
+		# check with something with non-zero len_L_mat_elements
+		dim = 8
+		mu_prior = np.zeros(dim)
+		prec_prior = np.diag(np.ones(dim) * 4)
+		mu_prop = mu_prior
+		prec_prop = prec_prior
+
+		gaussian_loss1 = Analysis.loss_functions.FullCovarianceLoss(dim)
+		snpe_c_loss1 = Analysis.loss_functions.FullCovarianceAPTLoss(dim,
+			mu_prior, prec_prior, mu_prop,prec_prop)
+
+		output1 = np.asarray([[  0.20284523,   0.29238915,  -0.7393373 ,  -0.03214657,
+         0.6250517 ,   0.2523826 ,   1.077436  ,  -0.69989514,
+         3.6308527 ,  23.381945  ,   2.2662773 ,  -4.85888   ,
+         3.962213  ,   3.1098747 ,   2.602856  ,   1.3564798 ,
+         3.3465786 ,   0.2168981 , -16.477966  ,  -8.322151  ,
+         2.487496  ,  -0.28127927,   1.1385951 ,   1.959164  ,
+        -4.360259  , -20.258226  ,  -0.56529176,  -0.7079977 ,
+         1.1026397 ,  -2.486014  ,   1.3936588 ,   6.8615203 ,
+         0.1961821 ,  -0.13286208,  -0.3439808 ,   1.2944454 ,
+         7.697075  ,   3.17325   ,  -7.014868  ,   0.30311686,
+         0.46607757,  -0.42691824,  -0.35380697,   1.5319805 ]])
+		truth1 = np.asarray([[ 0.33084044, -0.07505693, -0.7572751 , -0.90663636,  0.34479252,
+       -0.02982552,  1.0541298 , -0.32437885]])
+		output1 = tf.constant(output1,dtype=tf.float32)
+		output1_batched = tf.squeeze(tf.stack([output1,output1]))
+		truth1 = tf.constant(truth1,dtype=tf.float32)
+		truth1_batched = tf.squeeze(tf.stack([truth1,truth1]))
+
+		# prior = proposal, so should be same as gaussian loss
+		loss_diff = np.abs(gaussian_loss1.loss(truth1,output1).numpy()[0] -
```

Comment:
Probably better to do assert_almost_equal and just play with the precision the test requires